src/router/controller/params/decorators.ts

```diff
@@ -55,7 +79,32 @@ export const Body = (prop?: string) =>
 		if (!body) {
 			return null;
 		}
-		return prop ? body[prop] : body;
+
+		// Extract Class type of Parameter with @Body
+		const { parameterIndex } = opts;
+		const params: Constructor[] = MetadataHelper.getMetadata(
+			'design:paramtypes',
+			opts.target,
+			opts.propertyKey,
+		);
+
+		// Make the validation of body
+		if (isArray(params)) {
+			const validation = validateObject(body, params[parameterIndex]);
+			if (validation.length > 0) {
+				throw {
```

Comment:
You could probably use an `HttpException` class or derived

Reply:
Great! I make a class extending HttpException and put it together the others Exceptions Classes. Using 
```ts
export class NotValidBodyException<T> extends HttpException {
	reasons: T;
	constructor(reasons: T) {
		super(HTTP_STATUS.BAD_REQUEST, 'Body bad formatted');
		this.reasons = reasons;
	}
}
```